### DISCUSSION TITLE
Any environment variable for setting buildtrees directory globally?

### DISCUSSION THREAD

**Discussion by reitowo** [microsoft/vcpkg#29079] [Q&A]
Can I set some environment variables to set `VCPKG_INSTALL_OPTIONS` or `VCPKG_BUILDTREES_ROOT` globally? To make it easier for setting `--x-buildtrees-root` for every cmake project.

**Comment by autoantwort** [comment]
I think one problem is that the buildtrees root can not be used simultaneously by multiple vcpkg instances, but if you set it globally and run multiple vcpkg instances simultaneously you get hard to debug errors.

**Reply by reitowo** [reply]
Maybe a file lock just like `vcpkg install` does?

**Reply by autoantwort** [reply]
Yeah that would help. I think there was a plan some time ago to replace the global file lock with more fine granulated ones.
